### PR TITLE
add wait-for-it.sh

### PR DESCRIPTION
**Pull request by chrisns**
i'm making another pull request that depends on this, but it's handy to have since it can be used to make slaves wait for the master to be up before trying to start and connect

### CODE REVIEW

**Comment by dergraf** [comment]
Can you merge this one with #11 ?

File: Dockerfile [hunk 1]
```diff
@@ -6,14 +6,14 @@ RUN apt-get update && apt-get install -y \
     libssl-dev \
     logrotate \
     sudo \
-&& rm -rf /var/lib/apt/lists/*
+    && rm -rf /var/lib/apt/lists/*
 
 ENV VERNEMQ_VERSION 0.15.3
 
 ADD https://bintray.com/artifact/download/erlio/vernemq/deb/jessie/vernemq_$VERNEMQ_VERSION-1_amd64.deb /tmp/vernemq.deb
+ADD https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh /usr/sbin/wait-for-it.sh
```
Comment (dergraf) [inline Dockerfile]: I'd prefer to do a local ADD here

**Comment by chrisns** [comment]
Can do, #11 is a bigger work in progress and this could be used by other things like kubenetes

File: Dockerfile (hunk 1 above)
Comment (chrisns) [inline Dockerfile]: I was trying to avoid licence conflicts and stuff by pulling it in at build :)

Comment (codeadict) [inline Dockerfile]: I guess you can just use `nc -z HOST PORT` with retries:
```bash
is_up=0
MAX_TRIES=60
for (( i = 0; i <= MAX_TRIES; i ++ )); do
        debug "Waiting for MQTT port(${port}) to become available at ${host} ..."
        nc -z ${host} 1883
        if [ "$?" == "0" ]; then
            debug "${host} seems to be up!"
            is_up=1
            break
        fi
        debug "Waiting for ${host} to come up ${i}/${MAX_TRIES}"
        sleep 1
 done
```

Comment (chrisns) [inline Dockerfile]: yeah, that'd do it too, though wait-for-it.sh allows you to if you wanted to wait for all the listeners to come up. i don't know if thats necessarily a thing that you want; you could of course chain to get the same thing.
My current use case is https://github.com/erlio/docker-vernemq/pull/11/files#diff-e43a532580d1e894a963b6cbdf8c2c91R72

**Pull request closed without merging**